### ISSUE TITLE
Intellij suggests pub get, but doing so fails

### ISSUE THREAD

**Issue by leafpetersen** [flutter/flutter-intellij#558]
If you don't run `flutter update-packages` in a clean checkout of the flutter repo, and then open a directory in the flutter repo, Intellij suggests doing a pub get.  However, doing so fails with an error message.  Running `flutter update-packages` resolves the issue.  

Not sure if this should work, but the failure mode is a little surprising.

To reproduce:

```
git clone https://github.com/flutter/flutter.git
export PATH=`pwd`/flutter/bin:$PATH
flutter doctor && flutter doctor
```

Now open Intellij, and open the `flutter/packages/flutter`.  Enable dart support, and a suggestion pops up indicating that `pub get` has not been run.  Clicking `get dependencies` produces the following error:

```

Working dir: /Users/leafp/src/flutter/packages/flutter
/Users/leafp/src/flutter/bin/cache/dart-sdk/bin/pub get
Resolving dependencies...
The Flutter SDK is not available.
Depended on by:
- flutter 
Process finished with exit code 69
```

Screenshot attached.

<img width="1373" alt="screen shot 2016-12-15 at 3 35 43 pm" src="https://cloud.githubusercontent.com/assets/8484504/21246232/3fa96796-c2dc-11e6-81d1-9b24ea5d66fd.png">

cc @Hixie

**Comment by Hixie** [comment]
Based on that message I'd guess that what's happening is that the message is the Dart plugin using the Dart SDK, rather than the Flutter plugin doing whatever Flutter would do.

**Comment by leafpetersen** [comment]
> Based on that message I'd guess that what's happening is that the message is the Dart plugin using the Dart SDK, rather than the Flutter plugin doing whatever Flutter would do.

Possible.  This is a big point of confusion for me here:  I'm just opening up directories (e.g. `packages/flutter`) and it seems to "work".  The Dart SDK is correctly being set to the flutter one (rather than the one that I have configured for my other Dart projects, see screenshot), but I don't know if that is sufficient.

Should I be opening things differently (e.g. using the New Project workflow)?  That doesn't seem to work though: even if I agree to overwrite the existing flutter.iml file, I get this error:

```
Waiting for another flutter command to release the startup lock...
Cannot create a project within the Flutter SDK.
Target directory '/Users/leafp/src/flutter/packages/flutter' is within the Flutter SDK at '/Users/leafp/src/flutter'.
```

<img width="1110" alt="screen shot 2016-12-15 at 3 45 35 pm" src="https://cloud.githubusercontent.com/assets/8484504/21246472/d856f048-c2dd-11e6-8f17-9626d4c66aa1.png">

**Comment by Hixie** [comment]
That last message is just telling you you can't create a project inside your flutter checkout. You have to create it outside your flutter checkout.

(The "Waiting for another flutter command to release the startup lock..." message means you have some other flutter command running in the background and they are experiencing lock contention.)

**Comment by mit-mit** [comment]
@pq @devoncarew I thinnk this is https://github.com/flutter/flutter-intellij/issues/316 ?

**Comment by pq** [comment]
On the surface, it looks like the issue is that the Flutter plugin is not installed (or up to date).

@leafpetersen could you confirm?

Some setup details here: https://flutter.io/intellij-ide/

**Comment by leafpetersen** [comment]
The flutter plugin is installed, you can see it listed immediately below the Dart plugin in the screenshot above.  Version details:

```
Flutter • channel master • https://github.com/flutter/flutter.git Framework • revision 2546259a64 (18 hours ago) • 2016-12-15 15:05:15 Engine • revision 2f32e8054d Tools • Dart 1.21.0
```

I suspect that what might be happening is that this project is not recognized as a Flutter project?  But I don't know how to get into a state that the project is recognized as a Flutter project.  As I mentioned, using the New Project workflow to create a new Flutter project (as suggested in the docs) doesn't seem to work within the flutter repo.

**Comment by pq** [comment]
Ah, sorry.  I didn't look closely enough.  I think you're right that it's a detection problem.  The good news is we improved this and the project should get detected as being Fluttery in the plugin update we're shooting to push out shortly.

**Comment by devoncarew** [comment]
@pq, fixed?

**Comment by mit-mit** [comment]
I can still repro this even with the flutter plugin installed with the following steps:

Make sure everything is clean slate:
1. `cd <flutter sdk dir>`
1. `rm -rf examples`
1. `git checkout examples`
1. `rm -rf ~/.pub-cache/hosted/*`

Open IJ:
1. `cd examples\flutter_gallery`
1. `idea .`
1. Open `lib\main.dart` from the project view

=> pub get banner:
![screen shot 2017-01-12 at 13 15 28](https://cloud.githubusercontent.com/assets/13644170/21889451/3f5f8510-d8c9-11e6-8948-18cbd027d00f.png)

Clicking `pub get` no longer fails though, but shouldn't it say `'flutter packages get' has not been run' ?

**Comment by mit-mit** [comment]
Is this just https://github.com/flutter/flutter-intellij/issues/509 ?

**Comment by pq** [comment]
Yep.  Now that the other issues are resolved I think #509 covers what's left.
